app/components/speakers/SpeakersController.js

```diff
@@ -77,6 +87,43 @@ speakersModule.controller('SpeakerDialogController',
                 {field: 'Position', value: sdc.speaker.position});
         }
 
+        sdc.count = function(speaker, sessions){
+            var count = 0;
+            sessions.filter(function(session){
+                if (session.speakers.filter(function(sp){
+                          return sp.id == speaker.id;
+                 })
+                 .length > 0)
+                    count++;                            
```

Comment:
The filter function should not be used to do calculations inside. I suppose you are interested in counting the number of sessions by the particular speaker. Then you can just use:

```js
var matchingSessions = sessions.filter(function(session){return session.speakers.filter(function(sp){return sp.id == speaker.id}).length > 0;})

size = matchingSessions.length
```

You can always assume that you will get array back, the filter function must return `true` or `false` and should not do any calculation inside.